assembl/static/js/app/views/baseMessageList.js

```diff
@@ -0,0 +1,2366 @@
+'use strict';
+/**
+ * 
+ * @module app.views.baseMessageList
+ */
+
+var Backbone = require('backbone'),
+    Raven = require('raven-js'),
+    ObjectTreeRenderVisitor = require('./visitors/objectTreeRenderVisitor.js'),
+    objectTreeRenderVisitorReSort = require('./visitors/objectTreeRenderVisitorReSort.js'),
+    MessageFamilyView = require('./messageFamily.js'),
+    _ = require('underscore'),
+    $ = require('jquery'),
+    Assembl = require('../app.js'),
+    Ctx = require('../common/context.js'),
+    Message = require('../models/message.js'),
+    i18n = require('../utils/i18n.js'),
+    PostQuery = require('./messageListPostQuery.js'),
+    Permissions = require('../utils/permissions.js'),
+    MessagesInProgress = require('../objects/messagesInProgress.js'),
+    scrollUtils = require('../utils/scrollUtils.js'),
+    CollectionManager = require('../common/collectionManager.js'),
+    Promise = require('bluebird');
+
+
+/**
+ * Constants
+ */
+var MESSAGE_LIST_VIEW_STYLES_CLASS_PREFIX = "js_messageList-view-",
+/* The maximum number of messages that can be loaded at the same time
+ * before being removed from memory
+ */
+    MAX_MESSAGES_IN_DISPLAY = 50,
+/* The number of messages to load each time the user scrolls to
+ * the end or beginning of the list.
+ */
+    MORE_PAGES_NUMBER = 20,
+    SLOW_WORKER_DELAY_VALUE = 20;
+
+/**
+ * A mixin for message list behaviour. Logic here, UI in subclasses.
+ * Note: implementing the mixin as a class->class functor is inspired by
+ * http://justinfagnani.com/2015/12/21/real-mixins-with-javascript-classes/
+ * @class app.views.baseMessageList.BaseMessageListMixin
+ */
+var BaseMessageListMixin = function(cls) {
```

Comment:
I did not know about Behavior, so it's not a deliberate choice against it. Refactoring may be non trivial, and I vote it should not block merge.